### PR TITLE
feat: history timeline shows relative time, such as today and yesterday

### PR DESCRIPTION
**Pull request by akumatus**
### TL;DR
First, fixed an i18n issue in history panel. When the browser language is set to Chinese, and the AFFiNE application language is set to English, the language supposed to be English global. But now the language is a mixture of Chinese and English, which is obviously wrong.

Second, design a time formatter to convert timestamp into relative calendar date, such today and yesterday and so on. Long-ago edits will show the exact date like before.

### What changed?
- `new Intl.DateTimeFormat` with language option form  `document.documentElement.lang`
- Added `timestampToCalendarDate` function to convert timestamp into relative calendar date
- Updated unit tests
- Updated i18n copywriting

### How to test?
1. Open view history version
2. Check edit timeline

### CODE REVIEW

**Comment by akumatus** [comment]
* **#6864** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/6864?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @akumatus and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by pengx17 — APPROVED** [review]
[cool](https://app.graphite.dev//settings/meme-library?org=toeverything)

**Review by pengx17 — APPROVED** [review]

File: packages/frontend/core/src/utils/intl-formatter.ts [hunk 1]
```diff
@@ -1,19 +1,65 @@
 import dayjs from 'dayjs';
 
-const timeFormatter = new Intl.DateTimeFormat(undefined, {
-  timeStyle: 'short',
-});
+function createTimeFormatter() {
+  return new Intl.DateTimeFormat(window.document?.documentElement?.lang, {
```
Comment (Brooooooklyn) [inline packages/frontend/core/src/utils/intl-formatter.ts]: pass the `window.document?.documentElement?.lang` from the caller, so we can run it on SSR

Comment (akumatus) [reply packages/frontend/core/src/utils/intl-formatter.ts]: Thanks for your comment! Now I use `getI18n()?.language` instead of `window.document?.documentElement?.lang` and everything looks fine.

**Review by Brooooooklyn — APPROVED** [review]
